src/assets/ba_data/python/baplus/_rpc.py

```diff
@@ -0,0 +1,249 @@
+                    self.details = 'Playing on a server'
+                    if follow != 'Never':
+                        self.party_join = json.dumps(
+                            {'address': hi.address, 'port': hi.port}
+                        )
+                        if follow == 'Always':
+                            self.state = hi.name
+            elif follow != 'Never' and (
+                not self.last_accessibility_check
+                or bs.apptime() - self.last_accessibility_check >= 5
+            ):
+                self.last_accessibility_check = bs.apptime()
+                bs.app.classic.master_server_v1_get(
+                    'bsAccessCheck',
+                    {
+                        'port': bs.get_game_port(),
+                        'b': bs.app.env.engine_build_number,
+                    },
+                    callback=bs.WeakCall(self.set_ip),
+                )
+
+            session = bs.get_foreground_host_session()
+            if session and len(session.sessionplayers) >= 1:
+                self.party_size = [
+                    len(session.sessionplayers),
+                    session.max_players,
+                ]
+            if isinstance(session, bs.CoopSession):
+                self.state = 'Co-op'
+            if isinstance(session, bs.DualTeamSession):
+                self.state = 'Teams'
+            if isinstance(session, bs.FreeForAllSession):
+                self.state = 'FFA'
+            activity = bs.get_foreground_host_activity()
+            if isinstance(activity, bs.GameActivity):
+                self.details = activity.get_instance_display_string().evaluate()
+                self.show_start_time = True
+
+            if follow == 'Never':
+                self.party_join = None
+                self.fb_party_join = None
+                self.party_id = str(uuid.uuid4())
+
+            if (self.party_join or self.fb_party_join) and not self.party_size:
+                self.party_size = [3, 6]
+        except Exception:
+            pass
+
+    def set_ip(self, data: dict[str, Any] | None) -> None:
+        if (
+            data
+            and 'accessible' in data
+            and data['accessible']
+            and 'address' in data
+            and 'port' in data
+        ):
+            import bascenev1 as bs
+
+            follow = bs.app.config.get('Allow RPC Joins', 'Invite Only')
+            if bs.get_public_party_enabled():
+                self.fb_details = 'Hosting a public party'
+                if follow == 'Always':
+                    self.fb_state = bs.app.config.get('Public Party Name', '')
+            self.fb_party_join = json.dumps(
+                {'address': data['address'], 'port': data['port']}
+            )
+        else:
+            self.fb_state = None
+            self.fb_details = None
+            self.fb_party_join = None
+
+    def connect_rpc(self) -> None:
+        self.rpc = Presence('1167093983579738112')  # type: ignore
```

Comment:
how do you plan on storing the client id?

Reply:
Probably as a constant at the top of the file